RIOT-OS/RIOT
# drivers/ina220: Changed to comply with RIOT's driver design goals & SAUL adaption

### Contribution description
This PR contains some major changes to the INA220 power/current monitor driver

1. Rename to INA2XX, as the same driver works for (at least) the INA220 and the INA219, most likely for a bunch of other I2C power/current monitors of the same family
2. Brought the API in compliance with RIOT's driver API goals
    - Use standard RIOT driver initialization/configuration via `ina2xx_params_t` instead of custom API
    - Perform a device reset and connectivity test upon initialization as required by the design goals stated in [RIOT's driver guide][guide]
    - Convert the results directly to reasonable physical units  as required by the design goals stated in [RIOT's driver guide][guide]
3. Integrated the driver into SAUL

[guide]: https://github.com/RIOT-OS/RIOT/wiki/Guide:-Writing-a-device-driver-in-RIOT

### Testing procedure

You'll need an INA219 or INA220 device, ideally hooked up at I2C interface 0 with the address inputs set to zero and a shunt resistor of 100 mΩ, as then the default configuration will work out of the box. (This default config will work for the popular INA219 break out boards, e.g. the ones by Adafruit or any clone.)

The default config is optimized to measure power/current for high-power USB devices (500mA @ 5V) with a comfortable margin (up to 655mA and up to 16V). This would also work for 12V power supplies as are popular e.g. with the Arduino boards.

1. Flash `examples/saul` with `USEMODULE=ina2xx make BOARD=<foo> flash` and interact with the saul shell command
2. Flash `tests/driver_ina2xx` and check the output. The values shown should be plausible

### Issues/PRs references
None
